packages/docprovider-extension/src/filebrowser.ts

```diff
@@ -4,72 +4,88 @@
+  IEditorWidgetFactory,
+  FileEditorFactory
+} from '@jupyterlab/fileeditor';
 import { ILogger, ILoggerRegistry } from '@jupyterlab/logconsole';
-import { INotebookTracker } from '@jupyterlab/notebook';
+import {
+  INotebookTracker,
+  INotebookWidgetFactory,
+  NotebookWidgetFactory
+} from '@jupyterlab/notebook';
 import { ISettingRegistry } from '@jupyterlab/settingregistry';
 import { ITranslator, nullTranslator } from '@jupyterlab/translation';
 
-import { CommandRegistry } from '@lumino/commands';
-
 import { YFile, YNotebook } from '@jupyter/ydoc';
 
 import {
-  ICollaborativeDrive,
+  ICollaborativeContentProvider,
   IGlobalAwareness
 } from '@jupyter/collaborative-drive';
-import { IForkProvider, TimelineWidget, YDrive } from '@jupyter/docprovider';
+import {
+  IForkProvider,
+  TimelineWidget,
+  RtcContentProvider
+} from '@jupyter/docprovider';
 import { Awareness } from 'y-protocols/awareness';
 import { URLExt } from '@jupyterlab/coreutils';
 
-/**
- * The command IDs used by the file browser plugin.
- */
-namespace CommandIDs {
-  export const openPath = 'filebrowser:open-path';
-}
 const DOCUMENT_TIMELINE_URL = 'api/collaboration/timeline';
 
 const TWO_SESSIONS_WARNING =
   'The file %1 has been opened with two different views. ' +
   'This is not supported. Please close this view; otherwise, ' +
   'some of your edits may not be saved properly.';
 
-/**
- * The default collaborative drive provider.
- */
-export const drive: JupyterFrontEndPlugin<ICollaborativeDrive> = {
-  id: '@jupyter/docprovider-extension:drive',
-  description: 'The default collaborative drive provider',
-  provides: ICollaborativeDrive,
-  requires: [ITranslator],
-  optional: [IGlobalAwareness],
-  activate: (
-    app: JupyterFrontEnd,
-    translator: ITranslator,
-    globalAwareness: Awareness | null
-  ): ICollaborativeDrive => {
-    const trans = translator.load('jupyter_collaboration');
-    const drive = new YDrive(app.serviceManager.user, trans, globalAwareness);
-    app.serviceManager.contents.addDrive(drive);
-    return drive;
-  }
-};
+export const rtcContentProvider: JupyterFrontEndPlugin<ICollaborativeContentProvider> =
+  {
+    id: '@jupyter/docprovider-extension:content-provider',
+    description: 'The RTC content provider',
+    provides: ICollaborativeContentProvider,
+    requires: [ITranslator],
+    optional: [IGlobalAwareness],
+    activate: (
+      app: JupyterFrontEnd,
+      translator: ITranslator,
+      globalAwareness: Awareness | null
+    ): ICollaborativeContentProvider => {
+      const trans = translator.load('jupyter_collaboration');
+      const defaultDrive = (app.serviceManager.contents as ContentsManager)
+        .defaultDrive;
+      if (!defaultDrive) {
+        throw Error(
+          'Cannot initialize content provider: default drive property not accessible on contents manager instance.'
+        );
+      }
+      const registry = defaultDrive.contentProviderRegistry;
+      if (!registry) {
```

Comment:
Will that make `jupyter-collaboration` only compatible with JupyterLab 4.4 or higher?

Reply:
Yes, I would propose that, the next version of `jupyter-collaboration` will require JupyterLab 4.4.